src/pal/inc/pal.h

```diff
@@ -139,12 +139,24 @@ extern "C" {
 #endif
 
 #define MAX_PATH 260
-#define _MAX_PATH   260 /* max. length of full pathname */
+#define _MAX_PATH 260
 #define _MAX_DRIVE  3   /* max. length of drive component */
 #define _MAX_DIR    256 /* max. length of path component */
 #define _MAX_FNAME  256 /* max. length of file name component */
 #define _MAX_EXT    256 /* max. length of extension component */
 
+// In some Win32 APIs MAX_PATH is used for file names (even though 256 is the normal file system limit)
+// use _MAX_PATH_FNAME to indicate these cases
+#define MAX_PATH_FNAME MAX_PATH
+
+#if defined(PATH_MAX)
+#define MAX_LONGPATH   PATH_MAX  /* max. length of full pathname */
+#elif defined(MAXPATHLEN)
+#define MAX_LONGPATH   MAXPATHLEN
+#else
+#define MAX_LONGPATH   1024
+#endif
```

Comment:
I think that this should unconditionally defined MAX_LONGPATH as 1024. The existing conditional definitions will result in different values when pal.h is used together with system headers vs. when it is used without system headers - that is very hard to reason about.